localstack-core/localstack/services/events/provider.py

```diff
@@ -1809,6 +1810,10 @@ def _process_entry(
             return
 
         region, account_id = extract_region_and_account_id(event_bus_name_or_arn, context)
+
+        trace_id = populate_trace_id(context.request.headers)
```

Comment:
You can replace that with the context trace, see the main review comment.

```suggestion
        trace_id = context.trace_context["aws_trace_header"]
```